src/partition/FileUtils.cc

```diff
@@ -82,6 +84,67 @@ void InputFile::read(void *buf, off_t off, size_t sz) const {
     }
 }
 
+InputFileArrow::InputFileArrow(fs::path const &path, off_t blockSize) : InputFile(path), _fd(-1), _sz(-1) {
+    char msg[1024];
+    struct ::stat st;
+
+    m_batchReader = std::make_unique<lsst::partition::ParquetFile>(ParquetFile(path.string().c_str()));
+    arrow::Status status = m_batchReader->SetupBatchReader(blockSize);
+    if (!status.ok()) throw std::runtime_error("Could not setup Arrow recordbatchreader");
+
+    int fd = ::open(path.string().c_str(), O_RDONLY);
+
+    if (fd == -1) {
+        ::strerror_r(errno, msg, sizeof(msg));
+        throw std::runtime_error("open() failed [" + path.string() + "]: " + msg);
+    }
+    if (::fstat(fd, &st) != 0) {
+        ::strerror_r(errno, msg, sizeof(msg));
+        ::close(fd);
+        throw std::runtime_error("fstat() failed [" + path.string() + "]: " + msg);
+    }
+    _fd = fd;
+    _sz = st.st_size;
+}
+
+InputFileArrow::~InputFileArrow() {
+    char msg[1024];
+    if (_fd != -1 && ::close(_fd) != 0) {
+        ::snprintf(msg, sizeof(msg), "close() failed [%s]", _path.string().c_str());
+        ::perror(msg);
+        ::exit(EXIT_FAILURE);
```

Comment:
Printing an error is fine, but exiting the whole application may be too drastic, random libraries should not control how process exits. For error message it might be better to use lsst logging.

Reply:
Done